components/callbacks/executors.go

```diff
@@ -42,37 +42,53 @@ var retryable4xxErrorTypes = []int{
 	http.StatusTooManyRequests,
 }
 
-type CanGetNexusCompletion interface {
-	GetNexusCompletion(ctx context.Context) (nexus.OperationCompletion, error)
-}
+type (
```

Comment:
Please don't group types this way, it's very non standard in Go.
Go prefers putting the type definitions just above the implementation.